integration/e2e/e2e_test.go

```diff
@@ -914,3 +925,98 @@ func hashFile(file string) string {
+	URL  string `json:"url"`
+}
+
+func channelParticipationList(n *nwo.Network, o *nwo.Orderer, expectedChannels []string, systemChannel ...string) {
+	authClient, unauthClient := OrdererOperationalClients(n, o)
+	listChannelsURL := fmt.Sprintf("https://127.0.0.1:%d/participation/v1/channels", n.OrdererPort(o, nwo.OperationsPort))
+
+	By("using the channel participation API to list the channels")
+	body := getBody(authClient, listChannelsURL)()
+	list := &channelList{}
+	err := json.Unmarshal([]byte(body), list)
+	Expect(err).NotTo(HaveOccurred())
+
+	Expect(*list).To(MatchFields(IgnoreExtras, Fields{
+		"Channels":      channelsMatcher(expectedChannels),
+		"SystemChannel": systemChannelMatcher(systemChannel...),
+	}))
+
+	By("listing the channels without a client cert")
+	resp, err := unauthClient.Get(listChannelsURL)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(resp.StatusCode).To(Equal(http.StatusUnauthorized))
+
+	for _, channel := range list.Channels {
+		By("listing the details for channel " + channel.Name)
+		channelParticipationListOne(n, o, channel)
+	}
+}
+
+func channelsMatcher(channels []string) types.GomegaMatcher {
+	if len(channels) == 0 {
+		return BeEmpty()
+	}
+	matchers := make([]types.GomegaMatcher, len(channels))
+	for i, channel := range channels {
+		matchers[i] = channelInfoShortMatcher(channel)
+	}
+	return ConsistOf(matchers)
+}
+
+func systemChannelMatcher(systemChannel ...string) types.GomegaMatcher {
+	if len(systemChannel) == 0 {
+		return BeNil()
+	}
+	return PointTo(channelInfoShortMatcher(systemChannel[0]))
+}
+
+func channelInfoShortMatcher(channel string) types.GomegaMatcher {
+	return MatchFields(IgnoreExtras, Fields{
+		"Name": Equal(channel),
+		"URL":  Equal(fmt.Sprintf("/participation/v1/channels/%s", channel)),
+	})
+}
+
+type channelInfo struct {
+	Name            string `json:"name"`
+	URL             string `json:"url"`
+	Status          string `json:"status"`
+	ClusterRelation string `json:"clusterRelation"`
+}
+
+func channelParticipationListOne(n *nwo.Network, o *nwo.Orderer, channel channelInfoShort) {
+	authClient, _ := OrdererOperationalClients(n, o)
+	listChannelURL := fmt.Sprintf("https://127.0.0.1:%d/%s", n.OrdererPort(o, nwo.OperationsPort), channel.URL)
+
+	By("using the channel participation API to list the details of a single channel")
+	body := getBody(authClient, listChannelURL)()
+	c := &channelInfo{}
+	err := json.Unmarshal([]byte(body), c)
+	Expect(err).NotTo(HaveOccurred())
+
+	Expect(*c).To(MatchFields(IgnoreExtras, Fields{
+		"Name":            Equal(channel.Name),
+		"URL":             BeEmpty(),
+		"Status":          Equal("active"),
+		"ClusterRelation": Equal(clusterRelation(n.Consensus.Type)),
+	}))
+}
+
+// for the current test scenarios, we only have these two cluster relations.
+// other possible values are "follower" and "config-tracker"
+func clusterRelation(consensusType string) string {
+	if consensusType == "etcdraft" {
+		return "member"
+	}
+	return "none"
+}
```

Comment:
This is correct for these particular tests, but there is no re-use possible, so as soon as we hit "config-tracker" and "follower".... probably best not to embed this in the `*ListOne` method...